game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/EvaderRetreat.java

```diff
@@ -0,0 +1,141 @@
+package games.strategy.triplea.delegate.battle.steps.retreat;
+
+import games.strategy.engine.data.CompositeChange;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.changefactory.ChangeFactory;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import games.strategy.triplea.formatter.MyFormatter;
+import java.util.ArrayList;
+import java.util.Collection;
+import lombok.Builder;
+import lombok.experimental.UtilityClass;
+import org.triplea.sound.SoundUtils;
+
+@UtilityClass
+public class EvaderRetreat {
+
+  @Builder(toBuilder = true)
+  public static class Parameters {
+    BattleState battleState;
```

Comment:
nit, are the parameter values mutable? Can these be marked as final? It's perhaps also worthwhile to mark the non-null ones as `@Nonnull` so that they are checked when the builder is constructed. A downside of lombok builder is that it's not compile time safe, required args can be omitted, so generally it's important to at least get runtime validation with the `@Nonnull` annotation.